### PR TITLE
Initial baremetal server release from @JimmyMa

### PR DESCRIPTION
**Pull request by zhanggbj**

### CODE REVIEW

**Comment by zhanggbj** [comment]
Hi dr.max @maximilien,

Here is the Initial baremetal server release from @JimmyMa. Thanks.

File: README [hunk 1]
```diff
@@ -0,0 +1,6 @@
+## Baremaetal Provision Server
```
Comment (maximilien) [inline README]: Typo

File: jobs/baremetal-provision-server/spec [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    default: default_image
+  bps.port:
+    default: 8080
+    descript: "The server port"
+  bps.workers:
+    default: 6
+    descript: "The worker number"
+  bps.nfs:
+    default: ""
+    descript: "The  address of  NFS"
```
Comment (maximilien) [inline jobs/baremetal-provision-server/spec:34]: Why do we need NSF server?

File: jobs/xcat-server/templates/xcat_ctl.erb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+      apt-get -y --force-yes update
+      apt-get -y --force-yes install xcat
+      apt-get -y --force-yes install alien
+      apt-get -y --force-yes install cpanminus
+
+      alien -i xCAT-SoftLayer-*.rpm
+
+      export PERL_MM_USE_DEFAULT=1 # Disable the prompt in cpan
+#      perl -MCPAN -e 'install App::cpanminus'
```
Comment (maximilien) [inline jobs/xcat-server/templates/xcat_ctl.erb]: Remove commented code.

File: jobs/xcat-server/templates/xcat_ctl.erb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    netv6=`lsdef -t network | grep ":.*/64" | cut -d " " -f 1`
+    if [ -n "$netv6" ] ; then rmdef -t network $netv6 ; fi
+
+    cat /var/run/xcatd.pid > $PIDFILE
+
+    ;;
+
+  stop)
+     service xcatd stop
+#    kill_and_wait $PIDFILE
```
Comment (maximilien) [inline jobs/xcat-server/templates/xcat_ctl.erb]: Remove commented code

File: pre_create_release.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+git submodule update --init --recursive --force
+
+mkdir -p $ROOT_DIR/blobs/xcat/
+cd $ROOT_DIR/blobs/xcat/
+wget https://github.com/xcat2/xcat-core/releases/download/2.11_release/xcat-core-2.11-ubuntu.tar.bz2
+wget https://github.com/xcat2/xcat-core/releases/download/2.11_release/xcat-dep-ubuntu-2.11.tar.bz
+wget wget http://sourceforge.net/projects/xcat/files/yum/devel/core-snap/xCAT-SoftLayer-2.10-snap201507240527.noarch.rpm/download -O xCAT-SoftLayer-2.10-snap201507240527.noarch.rpm
+
+#wget http://ftp.usf.edu/pub/centos/6.7/isos/x86_64/CentOS-6.7-x86_64-netinstall.iso
```
Comment (maximilien) [inline pre_create_release.sh]: Remove commented code

**Comment by maximilien** [comment]
Address comments and I will merge. Thx

File: README (hunk 1 above)
Comment (zhanggbj) [inline README]: Done

File: jobs/baremetal-provision-server/spec (hunk 2 above)
Comment (zhanggbj) [inline jobs/baremetal-provision-server/spec:34]: Checked with Jimmy. This is used when to deploy baremetal server in an environment which crosses multiple VLANs. In this case, job (baremetal-provision-server and xcat-server) need to be deployed on each VLAN and they need to use NFS. If there are only one VLAN, no nfs is needed.

File: jobs/xcat-server/templates/xcat_ctl.erb (hunk 4 above)
Comment (zhanggbj) [inline jobs/xcat-server/templates/xcat_ctl.erb]: Done

File: pre_create_release.sh (hunk 5 above)
Comment (zhanggbj) [inline pre_create_release.sh]: Done.

**Comment by zhanggbj** [comment]
Comments Addressed. Thanks for review!

**Comment by maximilien** [comment]
Ok merging. This is a pretty complicated set of release ctl files. Hope you are testing this in pipeline

**Pull request merged**